components/camel-quartz/src/main/java/org/apache/camel/component/quartz/CamelJob.java

```diff
@@ -94,6 +101,18 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
         }
     }
 
+    private boolean hasTriggerExpired(JobExecutionContext context) {
+        Date fireTime = context.getFireTime();
+        if (context.getTrigger().getStartTime() != null && fireTime.before(context.getTrigger().getStartTime())) {
+            // Trigger invalid as Fire Time is before Start Time
+            return true;
+        } else if (context.getTrigger().getEndTime() != null && fireTime.after(context.getTrigger().getEndTime())) {
```

Comment:
I think this should not be an else if, as you can have set BOTH start and end time, and then you need to check for both of them.

Reply:
Indeed I missed the case, and corrected it in the new commit.